lightning/src/ln/channelmanager.rs

```diff
@@ -874,16 +874,74 @@ macro_rules! maybe_break_monitor_err {
 }
 
 macro_rules! handle_chan_restoration_locked {
-	($self: expr, $channel_lock: expr, $channel_state: expr, $channel_entry: expr,
-	 $raa: expr, $commitment_update: expr, $order: expr,
+	($self: ident, $channel_lock: expr, $channel_state: expr, $channel_entry: expr,
+	 $raa: expr, $commitment_update: expr, $order: expr, $chanmon_update: expr,
 	 $pending_forwards: expr, $funding_broadcastable: expr, $funding_locked: expr) => { {
 		let mut htlc_forwards = None;
 		let counterparty_node_id = $channel_entry.get().get_counterparty_node_id();
 
-		{
-			if !$pending_forwards.is_empty() {
+		let chanmon_update: Option<ChannelMonitorUpdate> = $chanmon_update; // Force type-checking to resolve
+		let chanmon_update_is_some = chanmon_update.is_some();
```

Comment:
Is this variable needed?

Reply:
Yes, inside the loop we move out of chanmon_update with an `if let Some(...) = `.

Reply:
Ah, interesting use of `loop`. Maybe name it `chanmon_update_is_none` instead of negating later. Or name it `has_chanmon_update`.